File: QID-3530-SFSportfolio/SFSportfolio.R
# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

x = read.table("Portf9206_logRet.dat")

n = nrow(x)
xf = apply(x, 2, sort)                          # Sort data in ascending order

t = (1:n)/(n + 1)
dat1 = cbind(pnorm((xf - mean(xf))/sd(xf)), t)  # Determine probabilities
dat2 = cbind(t, t)

# PP Plot
plot(dat1, col = "blue", ylab = (""), xlab = c(""))
lines(dat2, col = "red", lwd = 2.5)
title("PP Plot of Daily Return of Portfolio")

# QQ Plot
dev.new()
qqnorm(xf, col = "blue", xlab = c(""), ylab = c(""), main = ("QQ Plot of Daily Return of Portfolio"))
qqline(xf, col = "red", lwd = 2.5) 

File: QID-3530-SFSportfolio/SFSportfolio.m
% ---------------------------------------------------------------------
% Book:         SFS
% ---------------------------------------------------------------------
% Quantlet:     SFSportfolio
% ---------------------------------------------------------------------
% Description:  SFSportfolio produces a PP and a QQ Plot of the daily
%               log returns from 1992-01-01 to 2006-12-29 of a portfolio of
%               Bayer, BMW and Siemens stock.
% ---------------------------------------------------------------------
% Usage:        -
% ---------------------------------------------------------------------
% Inputs:       None
% ---------------------------------------------------------------------
% Output:       Normal PP-plot and normal QQ-plot of daily log-returns of 
%               portfolio (Bayer, BMW, Siemens).
% ---------------------------------------------------------------------
% Example:      -
% ---------------------------------------------------------------------
% Author:       Ying Chen 20030529
% ---------------------------------------------------------------------


close all;
clear all;
clc;

x    = load('Portf9206_logRet.dat');
n    = length(x);
xf   = sort(x);

t    = (1:n)/(n+1);
dat  = [normcdf((xf-mean(xf))/sqrt(var(xf))),t'];
dat2 = [t',t'];

hold on
figure(1)
scatter(dat(:,1),dat(:,2),'.','b')
plot(dat2(:,1),dat2(:,2),'r','LineWidth',2)
title('PP Plot of Daily Return of Portfolio')
hold off

figure(2)
h = qqplot(xf)
set(h(1),'Marker','.')
set(h(3),'LineStyle','-')
set(h(3),'LineWidth',2)
title('QQ Plot of Daily Return of Portfolio')
xlabel('')
ylabel('')

